### PR TITLE
chore: Change AMF to use references

### PR DESCRIPTION
**Pull request by jadebenn**
Yet another feature punched out from PR #1452. Part of the current pointer reduction obsess- er, I mean, push.

### CODE REVIEW

**Review by jadebenn — COMMENTED** [review]
ah, since I merged in the branch to get the changes, looks like I brought the history along with it. let me try and squash that...
nope. no good way to. oh well.

File: dGame/dPropertyBehaviors/ControlBehaviors.h [hunk 1]
```diff
@@ -19,15 +19,15 @@ class SystemAddress;
...
 		: arguments{ args }, modelComponent{ modelComponent }, modelOwner{ modelOwner } {
 	};
 
 	operator bool() const {
-		return arguments != nullptr && modelComponent != nullptr && modelOwner != nullptr;
+		return modelComponent != nullptr && modelOwner != nullptr;
 	}
 
-	AMFArrayValue* arguments;
+	std::reference_wrapper<const AMFArrayValue> arguments;
```
Comment (EmosewaMC) [inline dGame/dPropertyBehaviors/ControlBehaviors.h:30]: there a particular reason this cant be a reference?  the initializer list above should allow this correct?

**Review by EmosewaMC — DISMISSED** [review]
1 question on the functional object

File: dGame/dPropertyBehaviors/ControlBehaviors.h (hunk 1 above)
Comment (jadebenn) [reply dGame/dPropertyBehaviors/ControlBehaviors.h:30]: Using an actual const reference would make the struct non-moveable, only copyable, and a use of reference wrapper here doesn't require "get()," so it essentially has the semantic upside of representing non-nullability without any of the downsides of an actual reference

Comment (EmosewaMC) [reply dGame/dPropertyBehaviors/ControlBehaviors.h:30]: I think its fine to have this be non move able.  This should be created once per request anyways

Comment (jadebenn) [reply dGame/dPropertyBehaviors/ControlBehaviors.h:30]: Why change it? We're not even using .get() here. It's a perfect use case.

Comment (EmosewaMC) [reply dGame/dPropertyBehaviors/ControlBehaviors.h:30]: I'll trust your judgment here in that case then

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**